### PR TITLE
Add new developer guide

### PR DESCRIPTION
**Pull request by Fuzanwenn**
Add new developer guide.

### CODE REVIEW

**Review by wadia82 — APPROVED** [review]
LGTM

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -376,6 +376,51 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
       Use case resumes at step 2.
 
 
+
+**Use case: Add a bill to an appointment**
```
Comment (guokweijie) [inline docs/DeveloperGuide.md:380]: Adding

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -376,6 +376,51 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+
+* 3a. The format for addBill command is not followed.
+
+    * 3a.1 AddressBook shows an error message.
+
+      Use case resumes at step 2.
+
+
+**Use case: Edit a bill of an appointment**
```
Comment (guokweijie) [inline docs/DeveloperGuide.md:402]: Editing

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -376,6 +376,51 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+1. User requests to edit a bill of an appointment.
+2. AddressBook requests for details of the bill to be edited.
+3. User enters the requested details.
+4. AddressBook edits the bill.
+
+    Use case ends.
+
+
+**Extensions**
```
Comment (guokweijie) [inline docs/DeveloperGuide.md:414]: can add more extensions, such as when there is no such index of bill

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -376,6 +376,51 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+1. User requests to add a bill to an appointment.
+2. AddressBook requests for details of the bill to add to the appointment.
+3. User enters the requested details.
+4. AddressBook adds the bill to the appointment.
+
+    Use case ends.
+
+
+**Extensions**
```
Comment (guokweijie) [inline docs/DeveloperGuide.md:392]: You can add more extensions, such as when duplicate bills are added.

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -376,6 +376,51 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+2. AddressBook requests for details of the bill to be edited.
+3. User enters the requested details.
+4. AddressBook edits the bill.
+
+    Use case ends.
+
+
+**Extensions**
+
+* 3a. The format for editBill command is not followed.
```
Comment (TZL0) [inline docs/DeveloperGuide.md:416]: input format of EditBillCommand

**Review by TZL0 — APPROVED** [review]

**Pull request merged**